src/lib/currency/currency.ml

```diff
@@ -225,37 +224,30 @@ end = struct
 
   let length_in_triples = (length_in_bits + 2) / 3
 
+  [%%versioned
   module Stable = struct
     module V1 = struct
+      type t = Unsigned.t [@@deriving sexp, compare, hash]
+
+      let to_latest = Fn.id
+
       module T = struct
-        type t = Unsigned.t [@@deriving bin_io, sexp, compare, hash, version]
+        type typ = t [@@deriving sexp, compare, hash]
+
+        type t = typ [@@deriving sexp, compare, hash]
 
         let of_int = Unsigned.of_int
 
         let to_int = Unsigned.to_int
       end
 
-      include T
-      include Registration.Make_latest_version (T)
       include Codable.Make_of_int (T)
       include Hashable.Make (T)
       include Comparable.Make (T)
     end
+  end]
 
-    module Latest = V1
-
-    module Module_decl = struct
-      let name = "make_currency"
-
-      type latest = Latest.t
-    end
-
-    module Registrar = Registration.Make (Module_decl)
-    module Registered_V1 = Registrar.Register (V1)
-  end
-
-  (* bin_io, version omitted *)
-  (*  type t = Stable.Latest.t  [@@deriving sexp, compare, hash] *)
+  (* type t = Stable.Latest.t  [@@deriving sexp, compare, hash] *)
```

Comment:
Kill?

(Misplaced comment before, not sure what happened there..)

Reply:
I think the comment is there because we really shouldn't have `include Stable.Latest`, which contains the `bin_io` functions; it would be better to define the type, as in the comment, and get the other bits in via some other means. I'll try that.